daemon/graphdriver/driver.go

```diff
@@ -142,10 +142,36 @@ func GetDriver(name, home string, options []string) (Driver, error) {
 func New(root string, options []string) (driver Driver, err error) {
 	for _, name := range []string{os.Getenv("DOCKER_DRIVER"), DefaultDriver} {
 		if name != "" {
+			logrus.Infof("[graphdriver] trying provided driver %q", name) // so the logs show specified driver
 			return GetDriver(name, root, options)
 		}
 	}
 
+	// Guess for prior driver
```

Comment:
But I feel there is a case needing to allow bouncing back and forth... Not
blocking that. That is why it is Warn level log
On Apr 2, 2015 12:26 PM, "Tianon Gravi" <notifications@github.com> wrote:

> In daemon/graphdriver/driver.go
> <https://github.com/docker/docker/pull/11999#discussion_r27673154>:
>
> >  			return GetDriver(name, root, options)
> >  		}
> >  	}
> >
> > +	// Guess for prior driver
>
> I still think we should bail in the "guess" path if there's any ambiguity,
> and force the human to resolve it appropriately.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/11999/files#r27673154>.
>